bokeh/core/compat/__init__.py

```diff
@@ -24,3 +24,9 @@
 .. _mplexporter: https://github.com/mpld3/mplexporter/tree/master/mplexporter
 
 '''
+from ...util.deprecation import deprecated
+
+deprecated(
+    "MPL compatibility can no longer be successfully maintained, and is now deprecated. " +
+    "All MPL compat functions will be removed completely on the release of Bokeh 1.0. " +
```

Comment:
I might suggest saying "prior" instead of "on", because it might be good to remove them slightly ahead of 1.0 so that we can be sure nothing relies on them.